WikiDrips/WikiTableViewController.swift

```diff
@@ -76,26 +116,36 @@ class WikiTableViewController: UITableViewController, UISearchResultsUpdating {
         let wikiDoc = wikiDocs[indexPath.section][indexPath.row]
         wikiCell.wikiTitleLabel?.text = wikiDoc.title
         wikiCell.wikiDateLabel?.text = dateFormatter.string(from: wikiDoc.date)
-        
-        guard let initials = wikiDoc.imageInitials, let rectangle = wikiCell.wikiTitleImageView?.bounds else {
-            return wikiCell
-        }
-        
-        let cacheKey:NSString = initials as NSString //NSCache won't take String, so casting to NSString
-        
-        if let cachedImage = imageCache.object(forKey: cacheKey) {
-            wikiCell.wikiTitleImageView?.image = cachedImage
+        if let image = wikiDoc.image {
+            print("From WikiDoc")
+            wikiCell.wikiTitleImageView?.image = image
         } else {
-            DispatchQueue.global().async {
-                let generatedImage = UIImage.image(withInitials: initials, in: rectangle)
-                DispatchQueue.main.async() { [weak self] in
-                    wikiCell.wikiTitleImageView?.image = generatedImage
-                    self?.imageCache.setObject(generatedImage, forKey: cacheKey)
-                }
-            }
+            wikiCell.wikiTitleImageView?.image = nil
+            generateImage(forItemAtIndex: indexPath)
```

Comment:
Sorry, I'm confused again.  UITableViewDataSourcePrefetching methods only fire if there are cells which are not currently displayed.  If I comment out this line and if my wikiRequest comes back with only two WikiDocs, prefetching doesn't fire at all.  And those two docs never get images because they are always visible.